### DISCUSSION TITLE
Request for comment: Score parameters for reputation calculation

### DISCUSSION THREAD

**Discussion by HenrikJannsen** [bisq-network/bisq2#1063] [General]
I added a simulation tool to the different reputation sources to make it easier to reason about the right score parameters.

Here are the relevant screens with some different states:

## Burned BSQ:
<img width="1251" alt="Screenshot 2023-07-31 at 23 09 33" src="https://github.com/bisq-network/bisq2/assets/116298498/17553693-1550-47b3-8f3d-920991a49afe">
<img width="1251" alt="Screenshot 2023-07-31 at 23 09 50" src="https://github.com/bisq-network/bisq2/assets/116298498/ca651251-0185-4188-94b9-d9727946cba0">
<img width="1251" alt="Screenshot 2023-07-31 at 23 09 54" src="https://github.com/bisq-network/bisq2/assets/116298498/0b680bad-56c4-4bc4-9dfa-ef8dba05be53">

## Bonded BSQ:
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 01" src="https://github.com/bisq-network/bisq2/assets/116298498/7eaaedd5-e771-4b06-b215-64081d137992">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 05" src="https://github.com/bisq-network/bisq2/assets/116298498/db74de01-eb33-4b6e-a14e-a0eb62439ebc">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 08" src="https://github.com/bisq-network/bisq2/assets/116298498/4e3140d2-f8d3-41e4-a3e1-807bcb71d31f">

One need to consider that increasing the locktime to 100000 blocks (about 2 years) gives the same score as burning BSQ, but the effect of the age is still half compared to burning BSQ. I guess such a long locktime can be considered similar safe as burned BSQ.

## Account age:
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 14" src="https://github.com/bisq-network/bisq2/assets/116298498/4123f98b-20e2-481e-a397-5c2b32e62fd1">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 25" src="https://github.com/bisq-network/bisq2/assets/116298498/5c03847a-d2da-4a21-b140-d4b63d165373">

A 1 year old account has a score of 3650. The smallest amount which can be burned is 6 BSQ which results in a score of 6000 or 4380 after 1 year.
Maybe we could give it a bit more weight?

## Signed Witness:
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 30" src="https://github.com/bisq-network/bisq2/assets/116298498/cda1108e-70c4-4237-99d3-9862fcf1d216">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 36" src="https://github.com/bisq-network/bisq2/assets/116298498/ab240b55-023e-43da-81be-7b36b583a359">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 39" src="https://github.com/bisq-network/bisq2/assets/116298498/5a0b896b-08e3-4b05-a463-27e3084461c3">
<img width="1251" alt="Screenshot 2023-07-31 at 23 10 42" src="https://github.com/bisq-network/bisq2/assets/116298498/381421ff-8354-438c-b680-8ce82851b308">

One need to burn 18.25 BSQ to reach a score of 18250 which a signed witness account gets after 1 year. 
Maybe we could give it a bit more weight?

Any feedback or change requests to those suggested parameters?

**Comment by pazza83** [comment]
Thanks for posting. Look good, if a little complicated!

Here are my comments:

Is the score cumulative? Can someone have all 4? Can they include multiple account ages?

**Burned BSQ**

"Burning BSQ is considered the strongest form of reputation" - I would argue Bonding BSQ is a higher source of reputation as the user bonding BSQ has more to lose than the user that has burnt BSQ. Assuming a user was looking to gain reputation for the lowest cost to scam users they would likely choose to burn BSQ rather than bond BSQ. Maybe should have same wording as Bonded BSQ "strong form or reputation"

Despite this I think the BSQ burned should have a X10 weight than Bonded BSQ.

Why the linear decay to zero? Maybe it should be more binary Score lasts for X days then goes to zero.

**Bonded BSQ**

Again why the linear decay to zero? Maybe score lasts for period of time BSQ locked then goes to zero.

Maybe Bonds should be capped in the Bisq 1 software to 100000 blocks to prevent users from adding a zero and locking their BSQ for 20 years or longer!

**Account age**

I think account age should not be 5 times less than signed witness. Maybe it should be half? i.e 25

Why cap at one year? I think would be nice to give users with accounts of over 1 year some more benefit for trading on Bisq for a long time.

**Signed Witness:**

Again why cap at one year? I think would be nice to give users with accounts of over 1 year some more benefit for trading on Bisq for a long time.

**Comment by HenrikJannsen** [comment]
> a little complicated!

Yes, thats one concern I have. Its not easy to understand and to get an intuitive feeling for. Thats why I added the simulation tool.

Making it very simple would degrade the security and would make it harder to create incentives for certain behavior like burning bsq at a regular rate. E.g. a one time burning represents less security then a monthly burning as that reflects activity/work in addition to money. Work might be more of an deterrent to scammers than small amount of money.

> Is the score cumulative?

Yes. One can add any number of reputation sources and repeat the burn/bond bsq types. 
Account age and signed witness is only possible with one. The oldest account will be used in case there are multiple. 

> I would argue Bonding BSQ is a higher source of reputation as the user bonding BSQ has more to lose than the user that has burnt BSQ

The reasoning behind setting burned BSQ higher is that it is easier to lose it. It requires only the moderator to ban a user and the investment in the reputation is gone. 
For confiscating a bonded BSQ its much more effort and not guaranteed that it succeeds. It requires a high quorum in DAO voting. And the scammer could try to attack the DAO voting process to delay it until they can unlock the bond.

Another reason for the higher weight is that burning BSQ is the only way how Bisq can earn a fee from the BisqEasy trade protocol. Bonded BSQ do not provide that function. 

But yes we can improve the wording to make things more clear. Feel free to send me any suggestions.

> Despite this I think the BSQ burned should have a X10 weight than Bonded BSQ.

It has. Burned BSQ has weight 1000 and bonded BSQ 100. But its more complicate as the locktime plays a factor. With min. locktime of 10k blocks (70 days) its 10x (ignoring decay factor). But if the user increase locktime they get closer. A locktime of 100k (2 years) would result in the same score, but still the decay is different, so the burned BSQ gains more over time.

> Why the linear decay to zero? Maybe it should be more binary Score lasts for X days then goes to zero.

Hm... why do you think that is better? I try to avoid abrupt changes as it also might confuse user experience (e.g. you trade with a seller with high score and suddenly it goes to 0 versus you see the score decreasing over time which might spark interest of the buyer to learn about the details of the reputation system and maybe he becomes a seller at some point themself).

> Maybe Bonds should be capped in the Bisq 1 software to 100000 blocks to prevent users from adding a zero and locking their BSQ for 20 years or longer!

Yes good point. Or at least add a warning popup. Might be useful for other bonds as well. I think no hard limit leaves more flexibility. It turns a bonded BSQ more closer to burned BSQ (if you dont get back the money in some reasonable life time its similar as lost money).

> I think account age should not be 5 times less than signed witness. Maybe it should be half? i.e 25

Ok. Do you consider the weight of signed account in relation to burn/bond OK?

> Why cap at one year? I think would be nice to give users with accounts of over 1 year some more benefit for trading on Bisq for a long time.

Agree. What max time should we allow? We have to limit it as account age is not super secure and we dont want to create an incentive that someone tries to trick account age system to get a very high reputation. Also there might be users who have used Bisq many years ago but have not used it in the meantime. So that would not reflect an active user who can be trusted by long term activity without scams. It could be even a scammer who looked into Bisq long ago but never traded and giving them now a big boost for reputation might be risky.

Alternatively we could allow multiple account age and witness entries. That might reflect activity on Bisq better then age alone. I would need to look into it closer if there are any technical reasons why we could not allow multiple ones, but I guess not.

**Comment by pazza83** [comment]
Hopefully the complexity will not be too much an issue for the sellers on the platform. Better for the complexity to rest with the seller's (experienced users) than the buyers (new users).

> The reasoning behind setting burned BSQ higher is that it is easier to lose it. It requires only the moderator to ban a user and the investment in the reputation is gone.
> For confiscating a bonded BSQ its much more effort and not guaranteed that it succeeds. It requires a high quorum in DAO voting. And the scammer could try to attack the DAO voting process to delay it until they can unlock the bond.

Thanks for the additional info. It makes sense why you would consider Burned BSQ a stronger form or reputation than bonding.

> Hm... why do you think that is better? I try to avoid abrupt changes as it also might confuse user experience (e.g. you trade with a seller with high score and suddenly it goes to 0 versus you see the score decreasing over time which might spark interest of the buyer to learn about the details of the reputation system and maybe he becomes a seller at some point themself).

Makes sense with what you explained above for the decay to be linear. I was thinking in terms of making things more simple, but now I see why linear would be better.

> Ok. Do you consider the weight of signed account in relation to burn/bond OK?

Yes, will be interesting to see what reputation scores end up being. I imagine users with high reputation from bonding or burning BSQ will mean that sellers with only account age / signed witness as reputation will see less trades as a result. Creating an incentive for them to bond / burn BSQ so this is a good thing.

> Agree. What max time should we allow? 

Some account on Bisq are over 1000 days old. So maybe 3-5 years. As the reputation gained in small compared to bonding / burning I think the risk of a scammer with an old account being able to cheat the system is pretty low.

> Alternatively we could allow multiple account age and witness entries. That might reflect activity on Bisq better then age alone. I would need to look into it closer if there are any technical reasons why we could not allow multiple ones, but I guess not.

What about the trade history summary in > Portfolio > History > Show Summary
It gives a total of BSQ trade fees paid and BTC trade fees paid. Could this be hashed and then used as form or reputation?
